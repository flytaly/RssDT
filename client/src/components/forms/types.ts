import { ReactNode } from 'react';

export type FocusEvent = React.FocusEvent<HTMLInputElement | HTMLSelectElement>;

export interface CommonProps {
  IconSVG?: React.FunctionComponent<React.SVGAttributes<SVGElement>>;
  onFocus?: (event: FocusEvent) => void;
  onBlur?: (event: FocusEvent) => void;
  error?: string;
  title?: string;
  touched?: boolean;
  tag?: 'input' | 'select';
  children?: ReactNode;
}
