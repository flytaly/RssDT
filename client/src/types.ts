export enum DigestSchedule {
  // realtime = 'realtime',
  everyhour = 'everyhour',
  every2hours = 'every2hours',
  every3hours = 'every3hours',
  every6hours = 'every6hours',
  every12hours = 'every12hours',
  daily = 'daily',
}
export const DigestDisable = 'disable';

export enum ShareId {
  pocket = 'pocket',
  evernote = 'evernote',
  trello = 'trello',
}

export const periodNames: Record<DigestSchedule | 'realtime' | 'disable', string> = {
  realtime: 'realtime',
  everyhour: 'hourly',
  every2hours: '2-hourly',
  every3hours: '3-hourly',
  every6hours: '6-hourly',
  every12hours: '12-hourly',
  daily: 'daily',
  disable: 'disabled',
};
