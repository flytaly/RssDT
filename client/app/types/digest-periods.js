export const periodNames = {
    EVERYHOUR: '1 hour',
    EVERY2HOURS: '2 hours',
    EVERY3HOURS: '3 hours',
    EVERY6HOURS: '6 hours',
    EVERY12HOURS: '12 hours',
    DAILY: 'Daily',
};

export default {
    EVERYHOUR: 'EVERYHOUR',
    EVERY2HOURS: 'EVERY2HOURS',
    EVERY3HOURS: 'EVERY3HOURS',
    EVERY6HOURS: 'EVERY6HOURS',
    EVERY12HOURS: 'EVERY12HOURS',
    DAILY: 'DAILY',
};
