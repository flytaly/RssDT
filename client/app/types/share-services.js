const share = [{
    id: 'pocket',
    getUrl: url => `https://getpocket.com/edit?url=${encodeURIComponent(url)}`,
    iconUrl: '/static/share/pocket_32.png',
    title: 'Pocket',
}, {
    id: 'evernote',
    getUrl: (url, title) => `http://www.evernote.com/clip.action?url=${encodeURIComponent(url)}&title=${encodeURIComponent(title)}`,
    iconUrl: '/static/share/evernote_32.png',
    title: 'Evernote',
}, {
    id: 'trello',
    getUrl: (url, title) => `https://trello.com/en/add-card?url=${encodeURIComponent(url)}&name=${encodeURIComponent(title)}`,
    iconUrl: '/static/share/trello_32.png',
    title: 'Trello',
}];

module.exports = share;
