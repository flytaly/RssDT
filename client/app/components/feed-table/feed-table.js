/* eslint-disable react/prop-types */
import React, { useState, useEffect, useRef } from 'react';
import gql from 'graphql-tag';
import { useQuery, useMutation } from 'react-apollo-hooks';
import Link from 'next/link';
import get from 'lodash.get';
import ReactModal from 'react-modal';
import { Table, Th, Tr, Td, ButtonWithIcon } from './styled-table-parts';
import { DeleteButton, CancelButton } from '../styled/buttons';
import TrashIcon from '../../public/static/trash.svg';
import EditIcon from '../../public/static/edit.svg';
import EditFeedSidebar from './edit-feed-sidebar';
import { periodNames } from '../../types/digest-periods';
import { MY_FEEDS_QUERY, ME_QUERY } from '../../queries';
import RenderInBrowser from '../render-in-browser';

const DELETE_FEED_MUTATION = gql`mutation ($id: ID!){
  deleteMyFeed ( id: $id ) { id }
}`;

const formatDates = ({ createdAt, lastUpdate }, { timeZone, locale }) => {
    const createdAtDate = new Date(createdAt);
    const lastUpdateDate = new Date(lastUpdate);
    let localCreatedAt;
    let localLastUpdate;
    // On NextJS server use information from GraphQL to create Date strings with user timezone and locale
    if (!process.browser || !timeZone || !locale) {
        try {
            localCreatedAt = createdAtDate.toLocaleDateString(locale, { timeZone });
            localLastUpdate = lastUpdateDate.toLocaleString(locale, { timeZone });
        } catch (e) {
            console.error(e);
        }
    }
    if (!localCreatedAt) localCreatedAt = createdAtDate.toLocaleDateString();
    if (!localLastUpdate) localLastUpdate = lastUpdateDate.toLocaleString();

    return { localCreatedAt, localLastUpdate };
};

const renderRow = ({ feedInfo, meInfo, setConfirmDelete, setEditFeed }) => {
    const title = feedInfo.feed.title || feedInfo.feed.link || feedInfo.feed.url;
    const { localCreatedAt, localLastUpdate } = formatDates(feedInfo, meInfo);
    return (
        <Tr key={feedInfo.id}>
            <Td data-name="FEED">
                <Link href={`/feeds/view?id=${feedInfo.id}`}><a href={`/feeds/view?id=${feedInfo.id}`} title="View items">{title}</a></Link>
            </Td>
            <Td minWidth="8rem" data-name="ADDED">
                <RenderInBrowser>
                    {localCreatedAt}
                </RenderInBrowser>
            </Td>
            <Td data-name="LAST DIGEST DATE"><RenderInBrowser>{localLastUpdate}</RenderInBrowser></Td>
            <Td data-name="DIGEST SCHEDULE">{`${periodNames[feedInfo.schedule] || feedInfo.schedule} digest` }</Td>
            <Td minWidth="8rem" data-name="ACTIONS">
                <div>
                    <ButtonWithIcon
                        clickHandler={({ currentTarget }) => {
                            /* Firefox loses focus after clicking on button
                             hence ReactModal's 'shouldReturnFocusAfterClose' option doesn't work */
                            currentTarget.focus();

                            setEditFeed({
                                isOpen: true,
                                feedInfo,
                            });
                        }}
                        Icon={EditIcon}
                        title="Edit the feed"
                    />
                    <ButtonWithIcon
                        clickHandler={() => {
                            setConfirmDelete({
                                isOpen: true,
                                title,
                                id: feedInfo.id,
                            });
                        }}
                        Icon={TrashIcon}
                        title="Delete the feed"
                    />
                </div>
            </Td>
        </Tr>);
};

const updateAfterDeletion = (dataProxy, mutationResult) => {
    try {
        const { id } = mutationResult.data.deleteMyFeed;
        const data = dataProxy.readQuery({ query: MY_FEEDS_QUERY });
        data.myFeeds = data.myFeeds.filter(feed => feed.id !== id);
        dataProxy.writeQuery({ query: MY_FEEDS_QUERY, data });
    } catch (e) {
        console.error(e);
    }
};

const ResponsiveTable = () => {
    const { data: meData } = useQuery(ME_QUERY);
    const { data, loading, error } = useQuery(MY_FEEDS_QUERY);
    const [deleteFeedMutation] = useMutation(DELETE_FEED_MUTATION, { update: updateAfterDeletion });
    const [confirmDelete, setConfirmDelete] = useState({ isOpen: false, title: null, id: null });
    const [editFeed, setEditFeed] = useState({ isOpen: false, feedInfo: {} });
    const cancelBtnRef = useRef(null);
    useEffect(() => {
        ReactModal.setAppElement('body');
    }, []); // DOM doesn't exist on server so run setAppElement inside useEffect

    const feeds = get(data, 'myFeeds', []);
    const me = get(meData, 'me', {});
    if (error && error.message !== 'GraphQL error: Authentication is required') {
        console.error(error);
    }

    return (
        <>
            <Table>
                <Tr key="header">
                    <Th>FEED</Th>
                    <Th minWidth="8rem">ADDED</Th>
                    <Th>LAST DIGEST DATE</Th>
                    <Th>DIGEST SCHEDULE</Th>
                    <Th minWidth="8rem">ACTIONS</Th>
                </Tr>
                {loading ? 'loading...' : feeds.map(feedInfo => renderRow({ feedInfo, meInfo: me, setConfirmDelete, setEditFeed }))}
            </Table>
            <ReactModal
                isOpen={confirmDelete.isOpen}
                contentLabel="Confirm delete"
                shouldReturnFocusAfterClose={false}
                onRequestClose={() => setConfirmDelete({ isOpen: false })}
                onAfterOpen={() => cancelBtnRef.current.focus()}
                shouldCloseOnEsc
                shouldCloseOnOverlayClick
                style={{
                    overlay: {
                        display: 'flex',
                        justifyContent: 'center',
                        alignItems: 'center',
                    },
                    content: {
                        position: 'static',
                        textAlign: 'center',
                    },
                }}
            >
                <div style={{ marginBottom: '1rem' }}>
                    <span>Are you sure you want to delete the feed: </span>
                    <b>{confirmDelete.title}</b>
                    <span>?</span>
                </div>
                <CancelButton
                    style={{ marginRight: '1rem' }}
                    onClick={() => setConfirmDelete({ isOpen: false })}
                    ref={cancelBtnRef}
                >
                    Cancel
                </CancelButton>
                <DeleteButton
                    onClick={async () => {
                        setConfirmDelete({ isOpen: false });
                        const { id } = confirmDelete;
                        try {
                            await deleteFeedMutation({
                                variables: { id },
                                optimisticResponse: {
                                    __typename: 'Mutation',
                                    deleteMyFeed: {
                                        __typename: 'UserFeed',
                                        id,
                                    },
                                },
                            });
                        } catch (e) {
                            console.error(e);
                        }
                    }}
                >
                    Delete
                </DeleteButton>
            </ReactModal>
            <EditFeedSidebar editFeed={editFeed} setEditFeed={setEditFeed} />
        </>
    );
};

export default ResponsiveTable;
export { DELETE_FEED_MUTATION };
