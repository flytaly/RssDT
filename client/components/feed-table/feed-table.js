import React, { useState, useEffect, useRef } from 'react';
import gql from 'graphql-tag';
import { useQuery, useMutation } from 'react-apollo-hooks';
import get from 'lodash.get';
import ReactModal from 'react-modal';
import { Table, Th, Tr, Td, ButtonWithImg } from './styled-table-parts';
import { DeleteButton, CancelButton } from '../styled/buttons';
import trashIcon from '../../static/trash.svg';
import editIcon from '../../static/edit.svg';
import EditFeedSidebar from './edit-feed-sidebar';
import { periodNames } from '../../types/digest-periods';
import { MY_FEEDS_QUERY } from '../../queries';

const DELETE_FEED_MUTATION = gql`mutation ($id: ID!){
  deleteMyFeed ( id: $id ) { id }
}`;


const renderRow = (feedInfo, { setConfirmDelete, setEditFeed }) => {
    const title = feedInfo.feed.title || feedInfo.feed.link || feedInfo.feed.url;
    return (
        <Tr key={feedInfo.id}>
            <Td data-name="FEED">
                <a href={feedInfo.feed.url}>{title}</a>
            </Td>
            <Td minWidth="8rem" data-name="ADDED">
                {new Date(feedInfo.createdAt).toLocaleDateString()}
            </Td>
            <Td data-name="LAST DIGEST DATE">{new Date(feedInfo.lastUpdate).toLocaleString()}</Td>
            <Td data-name="DIGEST SCHEDULE">{`${periodNames[feedInfo.schedule] || feedInfo.schedule} digest` }</Td>
            <Td minWidth="8rem" data-name="ACTIONS">
                <div>
                    <ButtonWithImg
                        clickHandler={({ currentTarget }) => {
                            /* Firefox loses focus after clicking on button
                             hence ReactModal's 'shouldReturnFocusAfterClose' option doesn't work */
                            currentTarget.focus();

                            setEditFeed({
                                isOpen: true,
                                feedInfo,
                            });
                        }}
                        src={editIcon}
                        alt="Edit the feed"
                    />
                    <ButtonWithImg
                        clickHandler={() => {
                            setConfirmDelete({
                                isOpen: true,
                                title,
                                id: feedInfo.id,
                            });
                        }}
                        src={trashIcon}
                        alt="Delete the feed"
                    />
                </div>
            </Td>
        </Tr>);
};

const updateAfterDeletion = (dataProxy, mutationResult) => {
    try {
        const { id } = mutationResult.data.deleteMyFeed;
        const data = dataProxy.readQuery({ query: MY_FEEDS_QUERY });
        data.myFeeds = data.myFeeds.filter(feed => feed.id !== id);
        dataProxy.writeQuery({ query: MY_FEEDS_QUERY, data });
    } catch (e) {
        console.error(e);
    }
};

const ResponsiveTable = () => {
    const { data, loading, error } = useQuery(MY_FEEDS_QUERY);
    const deleteFeedMutation = useMutation(DELETE_FEED_MUTATION, { update: updateAfterDeletion });
    const [confirmDelete, setConfirmDelete] = useState({ isOpen: false, title: null, id: null });
    const [editFeed, setEditFeed] = useState({ isOpen: false, feedInfo: {} });
    const cancelBtnRef = useRef(null);
    useEffect(() => {
        ReactModal.setAppElement('body');
    }, []); // DOM doesn't exist on server so run setAppElement inside useEffect

    const feeds = get(data, 'myFeeds', []);
    if (error && error.message !== 'GraphQL error: Authentication is required') {
        console.error(error);
    }

    return (
        <React.Fragment>
            <Table>
                <Tr key="header">
                    <Th>FEED</Th>
                    <Th minWidth="8rem">ADDED</Th>
                    <Th>LAST DIGEST DATE</Th>
                    <Th>DIGEST SCHEDULE</Th>
                    <Th minWidth="8rem">ACTIONS</Th>
                </Tr>
                {loading ? 'loading...' : feeds.map(feedInfo => renderRow(feedInfo, { setConfirmDelete, setEditFeed }))}
            </Table>
            <ReactModal
                isOpen={confirmDelete.isOpen}
                contentLabel="Confirm delete"
                shouldReturnFocusAfterClose={false}
                onRequestClose={() => setConfirmDelete({ isOpen: false })}
                onAfterOpen={() => cancelBtnRef.current.focus()}
                shouldCloseOnEsc
                shouldCloseOnOverlayClick
                style={{
                    overlay: {
                        display: 'flex',
                        justifyContent: 'center',
                        alignItems: 'center',
                    },
                    content: {
                        position: 'static',
                        textAlign: 'center',
                    },
                }}
            >
                <div style={{ marginBottom: '1rem' }}>
                    <span>Are you sure you want to delete the feed: </span>
                    <b>{confirmDelete.title}</b>
                    <span>?</span>
                </div>
                <CancelButton
                    style={{ marginRight: '1rem' }}
                    onClick={() => setConfirmDelete({ isOpen: false })}
                    ref={cancelBtnRef}
                >
                    Cancel
                </CancelButton>
                <DeleteButton
                    onClick={async () => {
                        setConfirmDelete({ isOpen: false });
                        const { id } = confirmDelete;
                        try {
                            await deleteFeedMutation({
                                variables: { id },
                                optimisticResponse: {
                                    __typename: 'Mutation',
                                    deleteMyFeed: {
                                        __typename: 'UserFeed',
                                        id,
                                    },
                                },
                            });
                        } catch (e) {
                            console.error(e);
                        }
                    }}
                >
                    Delete
                </DeleteButton>
            </ReactModal>
            <EditFeedSidebar editFeed={editFeed} setEditFeed={setEditFeed} />
        </React.Fragment>
    );
};

export default ResponsiveTable;
export { DELETE_FEED_MUTATION };
