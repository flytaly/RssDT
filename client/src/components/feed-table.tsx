import Link from 'next/link';
import React, { useMemo, useState } from 'react';
import EditIcon from '../../public/static/edit.svg';
import TrashIcon from '../../public/static/trash.svg';
import { useDeleteMyFeedsMutation, UserFeed } from '../generated/graphql';
import { SortableColumn, useSortableState } from '../hooks/use-sortableState';
import { periodNames } from '../types';
import { updateAfterDelete as update } from '../utils/update-after-delete';
import CheckBox from './forms/checkbox';
import AddFeedModal from './modals/add-feed-modal';
import ConfirmModal from './modals/confirm-modal';
import EditFeedModal from './modals/edit-feed-modal';

type FeedTableProps = {
  feeds: Array<UserFeed>;
};

interface CellProps {
  children: React.ReactNode;
  name?: string;
  header?: boolean;
  className?: string;
}

const Cell: React.FC<CellProps> = ({ children, className = '', name = '', header = false }) => {
  return React.createElement(
    header ? 'th' : 'td',
    { className: `feed-table-column ${className}`, 'data-name': name },
    children,
  );
};

interface RowProps {
  children: React.ReactNode;
  isOdd?: boolean;
}
const Row: React.FC<RowProps> = ({ children, isOdd = true }) => (
  <tr
    className={`grid feed-table-template gap-2 py-4 sm:py-1 hover:bg-primary-2 ${
      isOdd ? '' : 'bg-gray-200'
    }`}
  >
    {children}
  </tr>
);

const HeaderRow: React.FC = ({ children }) => (
  <tr className="feed-table-template hidden sm:grid gap-2 font-bold  py-4 sm:py-1 hover:bg-primary-2">
    {children}
  </tr>
);

const ConfirmDeleteMsg: React.FC<{ feeds: UserFeed[]; error?: string }> = ({ feeds, error }) => (
  <div>
    Are you sure you want to delete the feeds:{' '}
    <ul className="max-h-32 overflow-auto text-sm">
      {feeds.map((uf) => (
        <li key={uf.id}>
          <b>{uf.title || uf.feed.title}</b>
        </li>
      ))}
    </ul>
    <div className="text-error">{error}</div>
  </div>
);

const formatDigestDate = (date?: string) => (date ? new Date(date).toLocaleString() : '-');
const formatCreatedDate = (date: string) => new Date(date).toLocaleDateString();

const FeedTable: React.FC<FeedTableProps> = ({ feeds }) => {
  const [feedsToDelete, setFeedsToDelete] = useState<UserFeed[]>([]);
  const [editingFeed, setEditingFeed] = useState<UserFeed | null>(null);
  const [deleteError, setDeleteError] = useState('');
  const [deleteMyFeeds, { loading }] = useDeleteMyFeedsMutation();
  const { getSortIcon, incSort, sortState, sortUserFeeds } = useSortableState();
  const [selectedIds, setSelectedIds] = useState<number[]>([]);
  const [addFeedModalOpen, setAddFeedModalOpen] = useState(false);

  const getHeaderButton = (col: SortableColumn, name: string) => (
    <button
      type="button"
      className="flex items-center font-bold w-full group"
      onClick={() => incSort(col)}
    >
      <span>{name}</span>
      {getSortIcon(col)}
    </button>
  );

  const closeModal = () => {
    setDeleteError('');
    setFeedsToDelete([]);
    setSelectedIds([]);
  };

  const sortedFeeds = useMemo(() => {
    if (!sortState.col || !sortState.dir) return feeds;
    return sortUserFeeds(feeds, sortState.col, sortState.dir);
  }, [feeds, sortState.col, sortState.dir, sortUserFeeds]);

  const onSelectChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const id = parseInt(e.target.id);
    return e.target.checked
      ? setSelectedIds((prevIds) => [...prevIds, id])
      : setSelectedIds((prevIds) => prevIds.filter(($id) => $id === id));
  };
  const toggleAll = () =>
    setSelectedIds((prev) => {
      if (prev.length) return [];
      return sortedFeeds.map((f) => f.id);
    });
  const removeMultipleFeeds = () => {
    const ids = new Set(selectedIds);
    setFeedsToDelete(sortedFeeds.filter((f) => ids.has(f.id)));
  };

  return (
    <div>
      <div className="flex justify-between items-end mb-2">
        {selectedIds.length ? (
          <span>
            <span>{`${selectedIds.length} selected feeds: `}</span>
            <button
              type="button"
              className="icon-btn inline-flex items-baseline px-1"
              onClick={removeMultipleFeeds}
            >
              <TrashIcon className="h-3 mr-1" />
              Remove
            </button>
          </span>
        ) : (
          <span />
        )}

        <button
          type="button"
          className="btn bg-secondary ml-auto"
          onClick={() => setAddFeedModalOpen(true)}
        >
          Add new feed
        </button>
      </div>
      {!sortedFeeds.length ? (
        <div className="text-center">You have no feeds</div>
      ) : (
        <table className="w-full text-sm">
          <thead>
            <HeaderRow>
              <Cell>
                <CheckBox
                  id="selectAll"
                  className="border-gray-400 hover:border-gray-600"
                  checked={selectedIds.length >= sortedFeeds.length}
                  onChange={toggleAll}
                />
              </Cell>
              <Cell name="Feed">{getHeaderButton('title', 'Feed')}</Cell>
              <Cell name="Added">{getHeaderButton('added', 'Added')}</Cell>
              <Cell name="Latest digest date">
                {getHeaderButton('digest_date', 'Latest digest date')}
              </Cell>
              <Cell name="Latest item pubdate">
                {getHeaderButton('item_pubdate', 'Latest item pubdate')}
              </Cell>
              <Cell name="Digest Schedule">Digest Schedule</Cell>
              <Cell name="Actions">Actions</Cell>
            </HeaderRow>
          </thead>
          <tbody>
            {sortedFeeds.map((uf, idx) => (
              <Row isOdd={!(idx % 2)} key={uf.id}>
                <Cell className="hidden sm:flex">
                  <CheckBox
                    id={`${uf.id}`}
                    className="border-gray-400 hover:border-gray-600"
                    checked={!!selectedIds.find((id) => id === uf.id)}
                    onChange={onSelectChange}
                  />
                </Cell>
                <Cell name="Feed">
                  <Link href={`/feed/${uf.id}`}>
                    <a className="underline">{uf.title || uf.feed.title}</a>
                  </Link>
                </Cell>
                <Cell className="text-xs" name="Added">
                  {formatCreatedDate(uf.createdAt)}
                </Cell>
                <Cell className="text-xs" name="Latest digest date">
                  {formatDigestDate(uf.lastDigestSentAt)}
                </Cell>
                <Cell className="text-xs" name="Latest item pubdate">
                  {formatDigestDate(uf.feed.lastPubdate)}
                </Cell>
                <Cell name="Digest Schedule">
                  {periodNames[uf.schedule] !== periodNames.disable ? (
                    <span className="font-medium">{periodNames[uf.schedule]}</span>
                  ) : (
                    <span className="text-gray-400">disabled</span>
                  )}
                </Cell>
                <Cell name="Actions">
                  <button onClick={() => setEditingFeed(uf)} type="button" className="icon-btn">
                    <EditIcon className="w-4 h-4" />
                  </button>
                  <button
                    onClick={() => setFeedsToDelete([uf])}
                    type="button"
                    className="icon-btn text-red-800"
                  >
                    <TrashIcon className="w-4 h-4 mr-1 " />
                  </button>
                </Cell>
              </Row>
            ))}
          </tbody>
        </table>
      )}
      <ConfirmModal
        isOpen={!!feedsToDelete.length}
        closeModal={closeModal}
        onConfirm={async () => {
          const ids = feedsToDelete.map(({ id }) => id);
          const result = await deleteMyFeeds({ variables: { ids }, update });
          if (result.errors?.length) setDeleteError("Couldn't delete feeds");
          else closeModal();
        }}
        disableButtons={loading}
        error={deleteError}
        message={<ConfirmDeleteMsg feeds={feedsToDelete} />}
      />
      <EditFeedModal
        feed={editingFeed}
        isOpen={!!editingFeed}
        closeModal={() => setEditingFeed(null)}
      />
      <AddFeedModal isOpen={addFeedModalOpen} closeModal={() => setAddFeedModalOpen(false)} />
    </div>
  );
};

export default FeedTable;
