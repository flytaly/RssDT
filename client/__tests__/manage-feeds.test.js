import { render, wait } from 'react-testing-library';
import ManageFeeds from '../components/manage-feeds';
import { MY_FEEDS_QUERY } from '../components/feed-table/feed-table';
import ApolloMockedProvider from '../test-utils/apollo-mocked-provider';
import types from '../types/digest-periods';

const mocks = [{
    request: { query: MY_FEEDS_QUERY },
    result: {
        data: {
            myFeeds: [{
                id: 'id1',
                createdAt: new Date('2019-04-19T00:11:00.000Z'),
                lastUpdate: new Date('2019-04-19T00:12:00.000Z'),
                schedule: types.DAILY,
                feed: {
                    title: 'Test Feed Title',
                    link: 'http://testfeed.com',
                    url: 'http://testfeed.com/feed',
                    imageTitle: 'Image Title',
                    imageUrl: 'http://testfeed.com/logo.png',
                },
            }],
        },
    },
}];

describe('Manage feeds', () => {
    const { container } = render(
        <ApolloMockedProvider mocks={mocks}>
            <ManageFeeds />
        </ApolloMockedProvider>,
    );

    test('should match snapshot', async () => {
        await wait(() => {
            expect(container.firstChild).toMatchSnapshot();
        });
    });
});
