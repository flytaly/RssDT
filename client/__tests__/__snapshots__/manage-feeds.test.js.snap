// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Manage feeds should match snapshot 1`] = `
.c2 {
  margin: 0px;
  padding: 0px;
  font-size: 1.3rem;
}

.c3 {
  list-style: none;
  display: grid;
  grid-template-columns: 6fr 2fr 4fr 4fr 1fr;
  grid-gap: 5px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.4rem 0;
}

.c3:hover:not(:first-of-type) {
  outline: 1px solid;
}

.c6 {
  min-width: 10rem;
  word-break: break-all;
}

.c7 {
  min-width: 8rem;
  word-break: break-all;
}

.c4 {
  font-size: 1.5rem;
  font-weight: bold;
  min-width: 10rem;
}

.c5 {
  font-size: 1.5rem;
  font-weight: bold;
  min-width: 8rem;
}

.c9 {
  height: 1.5rem;
}

.c8 {
  border: none;
  padding: 0;
  background: none;
  text-align: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin: 0 0.5rem 0 0;
}

.c8:focus,
.c8:hover {
  -webkit-transform: scale(1.10);
  -ms-transform: scale(1.10);
  transform: scale(1.10);
}

.c8:focus:hover {
  outline: none;
}

.c8:active {
  -webkit-transform: scale(0.95);
  -ms-transform: scale(0.95);
  transform: scale(0.95);
}

.c0 {
  padding: 0 2rem 1rem;
}

.c1 {
  font-size: 1.8rem;
}

.c10 {
  background-color: hsl(0,0%,60%);
  color: white;
  padding: 0.6rem 1rem;
  border: none;
  text-align: center;
  text-transform: uppercase;
  border-radius: 5px;
  min-width: 7rem;
  font-size: 1.3rem;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: hsl(200,100%,40%);
  margin-top: 1rem;
}

.c10:focus,
.c10:hover {
  background-color: hsl(0,0%,45%);
}

.c10:active {
  color: hsl(0,0%,90%);
}

.c10:focus,
.c10:hover {
  background-color: hsl(200,100%,30%);
}

.c10:active {
  color: hsl(200,100%,60%);
}

@media all and (max-width:) {
  .c3 {
    grid-template-columns: 1fr;
    padding: 1rem 0;
  }
}

@media all and (max-width:) {
  .c6 {
    display: grid;
    grid-template-columns: 40% 1fr;
    grid-gap: 5px;
  }

  .c6:before {
    content: attr(data-name);
    font-size: 1.5rem;
    font-weight: bold;
  }
}

@media all and (max-width:) {
  .c7 {
    display: grid;
    grid-template-columns: 40% 1fr;
    grid-gap: 5px;
  }

  .c7:before {
    content: attr(data-name);
    font-size: 1.5rem;
    font-weight: bold;
  }
}

@media all and (max-width:) {
  .c4 {
    display: none;
  }
}

@media all and (max-width:) {
  .c5 {
    display: none;
  }
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Edit subscriptions
  </h1>
  <ul
    class="c2"
  >
    <li
      class="c3"
    >
      <div
        class="c4"
      >
        FEED
      </div>
      <div
        class="c5"
      >
        ADDED
      </div>
      <div
        class="c4"
      >
        LAST DIGEST DATE
      </div>
      <div
        class="c4"
      >
        DIGEST SCHEDULE
      </div>
      <div
        class="c5"
      >
        ACTIONS
      </div>
    </li>
    <li
      class="c3"
    >
      <div
        class="c6"
        data-name="FEED"
      >
        <a
          href="/feeds/view?id=id1"
          title="View items"
        >
          Test Feed Title
        </a>
      </div>
      <div
        class="c7"
        data-name="ADDED"
      >
        4/19/2019
      </div>
      <div
        class="c6"
        data-name="LAST DIGEST DATE"
      >
        4/19/2019, 3:12:00 AM
      </div>
      <div
        class="c6"
        data-name="DIGEST SCHEDULE"
      >
        Daily digest
      </div>
      <div
        class="c7"
        data-name="ACTIONS"
      >
        <div>
          <button
            class="c8"
            type="button"
          >
            <img
              alt="Edit the feed"
              class="c9"
              src="/static/edit.svg"
            />
          </button>
          <button
            class="c8"
            type="button"
          >
            <img
              alt="Delete the feed"
              class="c9"
              src="/static/trash.svg"
            />
          </button>
        </div>
      </div>
    </li>
  </ul>
  <button
    class="c10"
    type="button"
  >
    Add new feed
  </button>
</main>
`;
